components/datadog/agent/docker.go

```diff
@@ -37,7 +39,7 @@ func (h *DockerAgent) Export(ctx *pulumi.Context, out *DockerAgentOutput) error
 }
 
 func NewDockerAgent(e config.CommonEnvironment, vm *remoteComp.Host, manager *docker.Manager, options ...dockeragentparams.Option) (*DockerAgent, error) {
-	return components.NewComponent(e, vm.Name(), func(comp *DockerAgent) error {
+	return components.NewComponent(e, vm.Name()+"-dockerAgent", func(comp *DockerAgent) error {
```

Comment:
`DockerAgent` is already in the type name, which is both visible in Pulumi output and in exported key.

Reply:
Removing it

Reply:
Wait, it's not visible in the `Host` name though, I don't understand how the name works then. Should I extract the name from the component type?

Reply:
Updated `NewComponent` doc to explicit how `name` is used with the reflected component name